shaders/constant_data/buffer_array.frag

```diff
@@ -0,0 +1,117 @@
+#version 320 es
+/* Copyright (c) 2019-2020, Arm Limited and Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+precision highp float;
+
+#ifdef HAS_BASE_COLOR_TEXTURE
+layout(set = 0, binding = 0) uniform sampler2D base_color_texture;
+#endif
+
+layout(location = 0) in vec4 in_pos;
+layout(location = 1) in vec2 in_uv;
+layout(location = 2) in vec3 in_normal;
+
+layout(location = 0) out vec4 o_color;
+
+struct MVPUniform
+{
+	mat4 model;
+	mat4 view_proj;
+
+#ifdef PUSH_CONSTANT_LIMIT_256
+	mat4 scale;
+	mat4 padding;
+#endif
+};
+
+layout(set = 0, binding = 1) uniform MVPUniformArray
```

Comment:
This seems to be the root cause of the issue on older Adreno devices. The compiler fails trying to create an 1800(ish) * 128 byte uniform.